vermouth/gmx/itp.py

```diff
@@ -83,7 +83,8 @@ def write_molecule_itp(molecule, outfile, header=()):
     """
     # Make sure the molecule contains the information required to write the
     # header.
-    _attr_has_not_none_attr(molecule, 'moltype')
+    if molecule.meta.get('moltype') is None:
+        raise ValueError('A molecule must have a moltype meta to write an ITP')
```

Comment:
Why does it? I disagree with it. Maybe just accept an optional name or moltype argument?

Reply:
You need a moltype to be specified somehow, or the ITP file that you will generate will be invalid. Same goes for the nrexcl. Everything else is optional. Of course, you could have an argument for moltype and nrexcl in `write_molecule_itp`, but 1) the idea is to have as much as possible defined in the molecule itself so it can be set with processors, and 2) how do you deal with conflicts between what is specified in the molecule and what is specified by arguments?

Reply:
Harrumph. Then I don't like moltype being demoted to a meta attribute. And this is rather Gromacs specific, so I'd say the attributes are optional (and should be in meta) and the processor should take them as argument. If None, take it from the molecule meta; otherwise, take the given argument.